pkg/search/postgres/common.go

```diff
@@ -268,31 +295,64 @@ func (p *parsedPaginationQuery) AsSQL() string {
 	return paginationSB.String()
 }
 
-func populateGroupBy(querySoFar *query, schema *walker.Schema, queryFields map[string]searchFieldMetadata) {
-	// Explicit GROUP BY clauses are not supported yet. If a query field (in select or order by) is a derived field requiring a group by clause,
+func populateGroupBy(querySoFar *query, groupBy *v1.QueryGroupBy, schema *walker.Schema, queryFields map[string]searchFieldMetadata) error {
+	// If explicit group by clauses are not specified and if a query field (in select or order by) is a derived field requiring a group by clause,
 	// default to primary key grouping. Note that all fields in the query, including pagination, are in `queryFields`.
-	for _, field := range queryFields {
-		if field.derivedMetadata == nil {
-			continue
+	if len(groupBy.GetFields()) == 0 {
+		for _, field := range queryFields {
+			if field.derivedMetadata == nil {
+				continue
+			}
+			switch field.derivedMetadata.DerivationType {
+			case searchPkg.CountDerivationType:
+				applyGroupByPrimaryKeys(querySoFar, schema)
+				return nil
+			}
+		}
+		return nil
+	}
+
+	for _, groupByField := range groupBy.GetFields() {
+		fieldMetadata := queryFields[groupByField]
+		dbField := fieldMetadata.baseField
+		if dbField == nil {
+			return errors.Errorf("field %s in GROUP BY clause does not exist in table %s or connected tables", groupByField, schema.Table)
+		}
+		if fieldMetadata.derivedMetadata != nil {
+			// Aggregate functions are not allowed in GROUP BY clause. SQL constraint.
+			return errors.Errorf("found %s in GROUP BY clause. Derived fields cannot be used in GROUP BY clause", groupByField)
+		}
+		if dbField.Options.PrimaryKey {
+			querySoFar.GroupByPrimaryKey = true
 		}
-		switch field.derivedMetadata.DerivationType {
-		case searchPkg.CountDerivationType:
-			applyGroupByPrimaryKeys(querySoFar, schema)
-			return
+
+		var cast string
+		if dbField.SQLType == "uuid" {
+			cast = "::text"
```

Comment:
Could you pls elaborate, why this cast is needed?

Reply:
Because we do not have uuid in go.